File: java/string-2/countCode.java
/* Return the number of times that the string "code" appears anywhere in the 
 * given string, except we'll accept any letter for the 'd', so "cope" and 
 * "cooe" count.
 */
public int countCode(String str) {
    int count = 0;
    
    for(int i = 0; i < str.length() - 3; i++) {
        if(str.substring(i, i + 2).equals("co") && str.charAt(i + 3) == 'e')
            count++;
    }
                    
    return count;
}

File: python/string-2/count_code.py
# Return the number of times that the string "code" appears anywhere in the 
# given string, except we'll accept any letter for the 'd', so "cope" and 
# "cooe" count.
def count_code(str):
    count = 0
    for i in range(len(str)-3):
        if str[i:i+2] == "co" and str[i+3] == "e":
            count += 1
                    
    return count
